src/linux/system.rs

```diff
@@ -786,24 +800,42 @@ enum InfoType {
 }
 
 #[cfg(not(target_os = "android"))]
-fn get_system_info(info: InfoType) -> Option<String> {
+fn get_system_info<T>(info: InfoType, buf_read: Option<T>) -> Option<String>
```

Comment:
Since the function is split between android and the rest, why having a `buf_read` here? I don't see the point... If you want to pass a different file to read, simply use `&Path` or `&str`.

Reply:
Android function was reverted so this was removed.

Reply:
Please don't pass a BufRead to the function. Just use a `&str` or a `&Path`.